src/compiler/checker.ts

```diff
@@ -1252,15 +1252,22 @@ namespace ts {
             }
         }
 
+        function SymbolLink(this: SymbolLinks) {

```

Comment:
This would be further improved by moving the constructor functions *outside* of `createTypeChecker`, since the same constructor function can be reused across multiple instances of the type checker when it's running as a language service.

Also, the name of the function should match the type:
```suggestion
        function SymbolLinks(this: SymbolLinks) {
```

Reply:
👍 will update this at home. Wasn’t aware you can shadow the name that way, but right, one is a Type, the other one is a Value.